tools/c7n_azure/c7n_azure/resources/storage_container.py

```diff
@@ -52,3 +55,49 @@ class resource_type(ChildTypeInfo):
         def extra_args(cls, parent_resource):
             return {'resource_group_name': parent_resource['resourceGroup'],
                     'account_name': parent_resource['name']}
+
+
+@StorageContainer.action_registry.register('set-public-access')
+class StorageContainerSetPublicAccessAction(AzureBaseAction):
+    """Action that updates the access level setting on Storage Containers.
+    Programmatically, this will be seen by updating the Public Access setting
+
+    :example:
+
+       Finds all Blob Storage Containers that are not private and sets them to private
+
+    .. code-block:: yaml
+
+        policies:
+            - name: set-non-production-accounts-private
+              resource: azure.storage-container
+              filters:
+                - type: value
+                  key: properties.publicAccess
+                  op: not-equal
+                  value: None
+              actions:
+                - type: set-public-access
+                  value: None
+    """
+    schema = type_schema(
+        'set-public-access',
+        required=['value'],
+        **{
+            'value': {'enum': ['Container', 'Blob', 'None']}
+        }
+    )
+
```

Comment:
Add the line `schema_alias = True` here under your schema.  (it is complicated :) )